src/store/mode/selectors.ts

```diff
@@ -0,0 +1,6 @@
+import { ModeTypes } from './types';
+import { ApplicationState } from '../types';
+
+const getAnnotationMode = (state: ApplicationState): ModeTypes => state.mode;
```

Comment:
Should we make the current mode a property on the mode state slice? So, `state.mode.current` or something similar?

Reply:
We could also consider using `reselect` for our selectors, since we get it for free with redux-toolkit.

Reply:
It looks like `reselect` is good for composing selectors or selectors for data derived from state that performs some sort of transformation. Could be something we can keep in mind for future development

Reply:
Sounds good.